pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -207,6 +207,26 @@ func (cm *certificateManager) GetOrCreateKeyPair(cli client.Client, secretName,
 	}, nil
 }
 
+// This type will be returned for errors that do not have the correct Key usage types
+// for a specific secert certificate.
+type CertUsageError struct {
+	msg string
```

Comment:
It would be nicer to build the error within the struct, since we have that capability. 

e.g., 

```
func NewCertUsage(name, ns string, requiredKeyUsages []x509.ExtKeyUsage) error {
  ...
}
```